lib/residence/residence_client_state_notifier.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_tutorial/residence/repository/residence_repository.dart';
+import 'package:flutter_tutorial/residence/state/residence_client_state.dart';
+
+final residenceClientStateNotifier = StateNotifierProvider.autoDispose<
+    ResidenceClientStateNotifier,
+    ResidenceClientState>((ref) => ResidenceClientStateNotifier(ref.read));
+
+class ResidenceClientStateNotifier extends StateNotifier<ResidenceClientState> {
+  ResidenceClientStateNotifier(this._read) : super(const ResidenceClientState()){
+    fetchResidenceItems();
+  }
+
+  final Reader _read;
+
+  Future<void> fetchResidenceItems() async {
+    state = state.copyWith(isLoading: true);
+    final residenceItems =
+    await _read(residenceRepositoryProvider).fetchResidenceItems();
+
+    if (residenceItems!.isNotEmpty) {
+      state = state.copyWith(
+        isLoading: false,
```

Comment:
`residenceItems`がnull許容型の場合は、念の為null checkをしたほうがいいですね！
```suggestion

    if (residenceItems != null && residenceItems.isNotEmpty) {
      state = state.copyWith(
        isLoading: false,
```

Reply:
上記のレビュー修正により「！」がいらなくなったので消しました！(元々参考にしたものも付いていかなかったので！)
ありがとうございます🙇‍♀️